shuup/campaigns/admin_module/form_sets.py

```diff
@@ -77,6 +80,27 @@ class BasketLineEffectsFormSet(EffectsFormset):
     def get_queryset(self):
         return self.owner.line_effects.instance_of(self._get_actual_model())
 
+    def clean(self):
```

Comment:
This validation shouldn't be here. It should be in `FreeProductLine.clean`. Don't forget to call super's clean too.